mdf.md

```diff
@@ -0,0 +1,75 @@
+#  MVDS Metadata Field
+
+> Version: 0.1.0 (Draft)
+> 
+> Authors: Oskar Thorén <oskar@status.im>, Dean Eigenmann <dean@status.im>
+
+##  Table of Contents
+
+1. [Abstract](#abstract)
+2. [Motivation](#motivation)
+3. [Format](#format)
+    1. [Fields](#fields)
+4. [Usage](#usage)
+    1. [`parents`](#parents)
+    2. [`ack_required`](#ack_required)
+
+## Abstract
+
+In this specification, we describe a method to construct message history that will aid the consistency guarantees of [MVDS](./mvds.md). Additionally we explain how data sync can be used for more lightweight messages that do not require full synchronization.
+
+## Motivation
+
+In order for more efficient synchronization of conversational messages, information should be provided allowing a node to more effectively synchronize the dependencies for any given message.
+
+## Format
+
+We introduce the metadata message which is used to convey information about a message and how it SHOULD be handled.
+
+```protobuf
+package vac.mvds;
+
+message Metadata {
+  bytes parents = 1;
+  bool ack_required = 2 [default = true];
+}
+```
+
+Nodes MAY transmit a `Metadata` message by extending the MVDS [message](./mvds.md#payloads) with a `metadata` field.
+
+```diff
+message Message {
++ Metadata metadata = 6000;
+  bytes group_id = 6001;
+  int64 timestamp = 6002;
+  bytes body = 6003;
+}
+```
+
+### Fields
+
+| Name                   |   Description                                                                                                                    |
+| ---------------------- | -------------------------------------------------------------------------------------------------------------------------------- |
+| `parents`               |   list of parent [`message indentifier`s](./mvds.md#payloads) for the specific message. |            
+| `ack_required`         |   indicates whether a message needs to be acknowledged or not.                                                             |
+
+## Usage
+
+### `parents`
+
+This field contains a list of parent [`message indentifier`s](./mvds.md#payloads) for the specific message. It MUST NOT contain any messages as parent whose `ack` flag was set to `false`. This creates a directed acyclic graph (DAG)<sup>1</sup> of persistent messages.
+
+This field helps establish consistency that may either be causal<sup>2</sup> or eventual<sup>3</sup> depending on the client.
```

Comment:
I would specify here that a client MAY buffer messages until depedencies are satisfied for causal consistency, otherwise MAY pass the messages straight away for eventual consistency.